### ISSUE TITLE
malformed raw string literal not captured

### ISSUE THREAD

**Issue by renkun-ken** [r-lib/lintr#892]
The syntax error of the raw string in the following code is not captured:

```r
line1 <- 1
line2 <- 2
line3 <- 3
line4 <- R"--/hello/--"
```

where error has the number of line:

```
malformed raw string literal at line 4
```

**Comment by MichaelChirico** [comment]
Could you elaborate on what's the error here?

Here's what I get at HEAD:

```
lintr::lint('line1 <- 1
line2 <- 2
line3 <- 3
line4 <- R"--/hello/--"')
# <text>:1:1: error: malformed raw string literal at line 4
# 
# ^
dput(.Last.value)
# structure(list(structure(list(filename = "<text>", line_number = 1L, 
#     column_number = 1L, type = "error", message = "malformed raw string literal at line 4", 
#     line = "", ranges = NULL, linter = NA_character_), class = "lint")), class = "lints")
```

Just that the `line_number` is incorrect?

**Comment by renkun-ken** [comment]
>Just that the line_number is incorrect?

Yes. Should be titled that the line number is not captured.
